server/routes/timelineResolver.ts

```diff
@@ -47,6 +47,14 @@ const eventsWithMergedCreateGoalEvents = (
         goalCreatedEvents.length > 1
           ? ({
               ...goalCreatedEvents[goalCreatedEvents.length - 1],
+              contextualInfo: goalCreatedEvents // Map all the goal created events contextualInfo properties into an object containing them all with an incremental number
+                .map((event, idx) =>
+                  idx > 0 ? event.contextualInfo : { GOAL_TITLE_0: event.contextualInfo.GOAL_TITLE },
+                )
+                .reduce((previous, current, idx) => ({
+                  ...previous,
+                  [`GOAL_TITLE_${idx}`]: current.GOAL_TITLE,
+                })),
```

Comment:
I may be misunderstanding this change, but I don't understand why the .map() is needed here – I think you might be able to just use a reduce?

```suggestion
              contextualInfo: goalCreatedEvents // Map all the goal created events contextualInfo properties into an object containing them all with an incremental number
                .reduce((previous, current, idx) => ({
                  ...previous,
                  [`GOAL_TITLE_${idx}`]: current.contextualInfo.GOAL_TITLE,
                }), {}),
```

Reply:
There are a couple of reasons;
Firstly, the reduce on its own returns an instance of the `previous` and `current` types - in this case it would have been a `TimelineEvent` (because we are iterating through an array of `goalCreatedEvents`). Actually what we need here is a new `contextualInfo` object, which is a simple key value pairs (`contextualInfo: { [key: string]: string }`)

Secondly, reduce is a funny old beast 🤣 in that the first iteration is actually for the 2nd element in the array, and the `previous` arg is the first element (ie. if there are 2 elements in the array, the `reduce` lambda will only be called once)
Because `previous` is the first element, and our objective is to create a object such as:
```
{
    GOAL_TITLE_0: '1st goal title`,
    GOAL_TITLE_1: '2nd goal title`,
    GOAL_TITLE_2: '3rd goal title`,
}
```
we need to "fudge" the first element to have a property name of `GOAL_TITLE_0`; otherwise the result is:
```
{
    GOAL_TITLE: '1st goal title`,
    GOAL_TITLE_1: '2nd goal title`,
    GOAL_TITLE_2: '3rd goal title`,
}
```